pkg/util/clientauthorizer/admin.go

```diff
@@ -5,23 +5,51 @@ package clientauthorizer
 
 import (
 	"crypto/tls"
-
-	"github.com/sirupsen/logrus"
+	"crypto/x509"
+	"errors"
+	"os"
 )
 
 type admin struct {
-	log *logrus.Entry
+	roots *x509.CertPool
+
+	expectedCommonName string
 }
 
-func NewAdmin(log *logrus.Entry) ClientAuthorizer {
-	return &admin{
-		log: log,
+func NewAdmin() (ClientAuthorizer, error) {
+	roots := x509.NewCertPool()
+	ok := roots.AppendCertsFromPEM([]byte(os.Getenv("ADMIN_CA_CERTS")))
```

Comment:
I'd like to get away from this being an environment variable, but maybe this will do for now.  Let's discuss quickly.